test/Jsr292/src/com/ibm/j9/jsr292/MethodHandleTest.java

```diff
@@ -1518,116 +1338,76 @@ public void test_isVarargsCollector_Using_asVarargsCollector_asTyped_arg_int_to_
 	public void testBindTo() throws Throwable {
         MethodHandle handle = SamePackageExample.getLookup().findSpecial(SamePackageExample.class, "isReceiverNull", MethodType.methodType(boolean.class), SamePackageExample.class);
         
-        AssertJUnit.assertFalse((boolean)handle.bindTo(new SamePackageExample()).invokeExact());
-        
-        boolean npeThrownWhenNullIsPassedInBindTo = false;
+        Assert.assertFalse((boolean)handle.bindTo(new SamePackageExample()).invokeExact());
         
-        try{
-        	boolean b = (boolean)(handle.bindTo(null).invokeExact());
-        }
-        catch(NullPointerException e) {
-        	npeThrownWhenNullIsPassedInBindTo = true;
-        }
-        AssertJUnit.assertTrue(npeThrownWhenNullIsPassedInBindTo);
+        boolean b = (boolean)(handle.bindTo(null).invokeExact());
+        Assert.fail("The test case failed to throw an NullPointerException in the case of null objects");
     }
     
     /**
      * Negative test for invoke
      * @throws Throwable
      */
-    @Test(groups = { "level.extended" })
+    @Test(expectedExceptions = { NumberFormatException.class, WrongMethodTypeException.class }, groups = { "level.extended" })
 	public void test_Invoke() throws Throwable {
     	MethodHandle handle = MethodHandles.lookup().findStatic(Integer.class, "parseInt", MethodType.methodType(int.class,String.class));
-    	boolean nfeHit = false;
-    	boolean wmtHit = false;
     	
-    	try {
-    		AssertJUnit.assertEquals(5,handle.invoke("s"));
-    	}
-    	catch(NumberFormatException e) {
-    		nfeHit = true;
-    	}
+    	Assert.assertEquals(handle.invoke("s"), 5);
+    	Assert.fail("The test case failed to throw an NumberFormatException");
     	
-    	AssertJUnit.assertTrue(nfeHit);
-    	
-    	try {
-    		AssertJUnit.assertEquals(5,handle.invoke(5));
-    	}
-    	catch(WrongMethodTypeException e) {
-    		wmtHit = true;
-    	}
-    	
-    	AssertJUnit.assertTrue(wmtHit);
+    	Assert.assertEquals(handle.invoke(5), 5);
+    	Assert.fail("The test case failed to throw an WrongMethodTypeException");
+
     }
     
 	/**
 	 * Negative test for invokeExcat
 	 * @throws Throwable
 	 */
-	@Test(groups = { "level.extended" })
+	@Test(expectedExceptions = { WrongMethodTypeException.class, WrongMethodTypeException.class }, groups = { "level.extended" })
 	public void test_InvokeExact() throws Throwable {
 		MethodHandle mh  = MethodHandles.lookup().findVirtual(SamePackageExample.class,"arrayToString",MethodType.methodType(String.class,Object[].class));
 		SamePackageExample g = new SamePackageExample();
 		mh = mh.bindTo(g);
-		boolean wmtHit = false;
-		try{
-			AssertJUnit.assertEquals("[a,b]",(String)mh.invokeExact("a","b"));
-		}
-		catch(WrongMethodTypeException e) {
-			wmtHit = true;
-		}
 		
-		AssertJUnit.assertTrue(wmtHit);
+		Assert.assertEquals((String)mh.invokeExact("a","b"), "[a,b]");
+		Assert.fail("The test case failed to throw an WrongMethodTypeException");
```

Comment:
Everything after this is dead code.  Can you split it into another test?

Reply:
Already split.